src/App.js

```diff
@@ -0,0 +1,20 @@
+import React, { Fragment, Component } from 'react'
+import Welcome from './components/welcome.js'
+
+export default class App extends Component {
+  constructor (props) {
+    super(props)
+
+    this.state = {
+      data: props.data
+    }
+  }
+
+  render () {
+    return (
+      <Fragment>
+        <Welcome name={this.state.data.name} />
```

Comment:
You can pass data directly using props instead of state since you are not making any state changes.